### DISCUSSION TITLE
Episode 166: Top 20 GPU SDKs, Libraries and Tools!

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#58] [Episode]
Episode Link: https://adspthepodcast.com/2024/01/26/Episode-166.html

Feel free to write comments or ask questions 🙂

**Comment by dholmes215** [comment]
Bryce's bit about the process failure with the plane QA was absolutely interesting!

**Comment by robbiesri** [comment]
(hella late as I work thru the backlog!)

I think this list is peak LLM just glomming together text answers from random articles without enough context. On the other hand, most of these articles don't know what they are talking about either.

The CUDA surface is interesting because NVIDIA ships layers and layers of CUDA infrastructure to support devs. Even the most baseline component is still the CUDA Toolkit, which encompasses the headers, runtime, samples, baseline libraries, and tools. Some users out there use the redistributables to get piecemeal components, but I'm guessing (aka _feeling_ hehe) that most users just get the entire toolkit.

Another point of order is that most the elements on the list are standards, not actual implementations with developer oriented SDKs. OpenGL/Vulkan especially. Depending on the platform, your dev experience is going to vary significantly. Vulkan on Stadia (the defunct game streaming platform from Google) was 100x better than Vulkan on Windows.  DX on Xbox is an entirely different beast than DX on Windows (though the Windows experience is quite good).

---

I don't know the timing, but there is this NVIDIA Enterprise Advisory Committee: https://www.nvidia.com/en-us/data-center/enterprise-platorm-advisor-program/

Maybe this was after the episode? In the game dev world (where I am from), platform owners have regular engagement with their dev communities. But they are an order of magnitude (or two) smaller than the CUDA surface.
